### PR TITLE
[TAS-24]✨ Add social feeds page

### PR DESCRIPTION
**Pull request by AuroraHuang22**

### CODE REVIEW

File: src/store/modules/wallet.js [hunk 1]
```diff
@@ -220,8 +230,10 @@ const getters = {
   walletInteractedCreators: state => state.pastFollowees,
   walletIsFetchingFollowees: state => state.isFetchingFollowees,
   walletIsFetchingFollowers: state => state.isFetchingFollowers,
+  walletIsFetchingFolloweeEvents: state => state.isFetchingFolloweeEvent,
   getIsFetchingEvent: state => state.isFetchingEvent,
   getEvents: state => state.events.slice(0, WALLET_EVENT_LIMIT),
+  getFolloweeEvents: state => state.followeeEvents.slice(0, 30),
```
Comment (williamchong) [inline src/store/modules/wallet.js:236]: please name the number 30 as a const

File: src/store/modules/wallet.js [hunk 2]
```diff
@@ -413,6 +425,119 @@ const actions = {
     }
   },
 
+  async fetchFolloweeWalletEvent({ state, commit, dispatch }) {
```
Comment (williamchong) [inline src/store/modules/wallet.js:428]: Can we reuse this function/store in notification page?

File: src/store/modules/wallet.js [hunk 3]
```diff
@@ -413,6 +425,119 @@ const actions = {
...
+          event,
+          collectEvents.find(collectEvent => collectEvent === event)
+        );
+      }
+    });
+
+    // filter available events
+    const filteredEvents = categorizedEvents.filter(event => {
+      if (event.sender === address || event.receiver === address) {
+        return false;
```
Comment (williamchong) [inline src/store/modules/wallet.js:526]: Why filter receiver is self events?

File: src/components/SocialFeedItem.vue [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+      return this.memo || this.granterMemo;
+    },
+    shouldShowItemCard() {
+      return this.type === EVENT_TYPE.PUBLISH;
+    },
+    shouldShowReceiverInfo() {
+      return this.type !== EVENT_TYPE.PUBLISH;
+    },
+    shouldShowFollow() {
+      return !this.walletFollowees.includes(this.senderWallet);
```
Comment (williamchong) [inline src/components/SocialFeedItem.vue:260]: In what case we will see a sender that we haven't followed?

Comment (AuroraHuang22) [reply src/components/SocialFeedItem.vue:260]: someone sent/collect NFT to my followee

File: src/store/modules/wallet.js (hunk 3 above)
Comment (AuroraHuang22) [reply src/store/modules/wallet.js:526]: We can currently see events where the recipient is yourself from the [notifications /]

File: src/store/modules/wallet.js (hunk 2 above)
Comment (AuroraHuang22) [reply src/store/modules/wallet.js:428]: Sure, but I prefer to do it in another PR 🌚.

File: src/components/NFTPortfolio/Item.vue [hunk 5]
```diff
@@ -97,6 +97,12 @@ export default {
         params: { classId: this.classId },
       });
     },
+    isSocialFeedsWithCollectiblesView() {
```
Comment (nwingt) [inline src/components/NFTPortfolio/Item.vue:100]:
```suggestion
    isSocialFeedWithCollectiblesView() {
```

File: src/components/NFTPortfolio/UserInfo.vue [hunk 6]
```diff
@@ -35,6 +35,9 @@
       </Label>
     </template>
     <slot name="stats" />
+    <div class="mt-[24px]">
```
Comment (nwingt) [inline src/components/NFTPortfolio/UserInfo.vue:38]:
```suggestion
    <div v-if="$slots.follow" class="mt-[24px]">
```

File: src/locales/en.json [hunk 7]
```diff
@@ -434,6 +434,14 @@
   "nft_portfolio_page_state_royalties": "Total Royalty\n(LIKE)",
   "nft_portfolio_page_state_sales": "Total Sales\n(LIKE)",
   "nft_portfolio_page_state_value": "Asset Value\n(LIKE)",
+  "nft_primitive_disclaimer": "The trading rule of Liker Land’s Writing NFT is not applicable to this NFT as it is published by a third-party.",
+  "nft_primitive_disclaimer_portfolio": "The trading rule of Liker Land’s Writing NFT is not applicable to these NFT as they are published by a third-party.",
+  "nft_primitive_nft_book": "The trading rule of this NFT Book is decided by its publisher.",
```
Comment (nwingt) [inline src/locales/en.json:439]: Are these keys not used anymore?

File: src/locales/zh-Hant.json [hunk 8]
```diff
@@ -425,6 +425,14 @@
   "nft_portfolio_page_state_royalties": "分潤收益 (LIKE)",
   "nft_portfolio_page_state_sales": "總銷售額 (LIKE)",
   "nft_portfolio_page_state_value": "資產值 (LIKE)",
+  "nft_primitive_disclaimer": "以下 NFT 由第三方發行，不適用於 Liker Land 的 Writing NFT 交易機制。",
+  "nft_primitive_disclaimer_portfolio": "以下 NFT 由第三方發行，不適用於 Liker Land 的 Writing NFT 交易機制。",
+  "nft_primitive_nft_book": "NFT 書的交易機制由發行者決定",
```
Comment (nwingt) [inline src/locales/zh-Hant.json:430]: Are these keys not used anymore?

File: src/mixins/wallet.js [hunk 9]
```diff
@@ -32,6 +32,8 @@ export default {
       'walletIsMatchedSession',
       'walletIsLoggingIn',
       'loginAddress',
+      'getFolloweeEvents',
+      'walletIsFetchingFolloweeEvents',
```
Comment (nwingt) [inline src/mixins/wallet.js:36]: If these computed props are only used in certain components, we might not want to add to wallet mixins, as the computed props update, those props will trigger unnecessary rendering

File: src/pages/_id/index.vue [hunk 10]
```diff
@@ -183,6 +234,12 @@ import walletMixin from '~/mixins/wallet';
 import portfolioMixin, { tabOptions } from '~/mixins/portfolio';
 import alertMixin from '~/mixins/alert';
 
+const FOLLOW_PROMPT_STATE = {
```
Comment (nwingt) [inline src/pages/_id/index.vue:237]: We should refactor this constant to `constants/index.js`

File: src/pages/social-feeds/index.vue [hunk 11]
```diff
@@ -0,0 +1,545 @@
...
+    handleGoCreated() {
+      this.handleCollectiblesTabChange(tabOptions.created);
+    },
+    handleCurrentTabChange(tab) {
+      switch (tab) {
+        case TAB_OPTIONS.TOWN:
+          logTrackerEvent(
+            this,
+            'SocialFeed',
+            'go_town_tab_clicked',
```
Comment (nwingt) [inline src/pages/social-feeds/index.vue:429]: For shorter name, we switched back to camelCase event name
```suggestion
            'FeedTabTownClick',
```

File: src/pages/social-feeds/index.vue [hunk 12]
```diff
@@ -0,0 +1,545 @@
...
+            this.wallet,
+            1
+          );
+          break;
+
+        case TAB_OPTIONS.COLLECTIBLES:
+          logTrackerEvent(
+            this,
+            'SocialFeed',
+            'go_collectibles_tab_clicked',
```
Comment (nwingt) [inline src/pages/social-feeds/index.vue:439]:
```suggestion
            'FeedTabCollectiblesClick',
```

File: src/pages/social-feeds/index.vue [hunk 13]
```diff
@@ -0,0 +1,545 @@
...
+            1
+          );
+
+          break;
+
+        case TAB_OPTIONS.USER_STATS:
+          logTrackerEvent(
+            this,
+            'SocialFeed',
+            'go_user_stats_tab_clicked',
```
Comment (nwingt) [inline src/pages/social-feeds/index.vue:450]:
```suggestion
            'FeedTabStatsClick',
```

File: src/components/SocialFeedItem.vue [hunk 14]
```diff
@@ -0,0 +1,407 @@
...
+          return this.$t('nft_social_feed_item_type_send');
+
+        case EVENT_TYPE.PUBLISH:
+          return this.$t('nft_social_feed_item_type_publish');
+
+        default:
+          return undefined;
+      }
+    },
+    formatTime() {
```
Comment (nwingt) [inline src/components/SocialFeedItem.vue:270]:
```suggestion
    formattedTime() {
```

File: src/components/SocialFeedItem.vue [hunk 15]
```diff
@@ -0,0 +1,407 @@
...
+    receiverId() {
+      return this.receiverInfo?.displayName;
+    },
+    senderAvatar() {
+      return this.senderInfo?.avatar;
+    },
+    receiverAvatar() {
+      return this.receiverInfo?.avatar;
+    },
+    formatType() {
```
Comment (nwingt) [inline src/components/SocialFeedItem.vue:254]:
```suggestion
    formattedType() {
```

File: src/components/SocialFeedItem.vue [hunk 16]
```diff
@@ -0,0 +1,407 @@
...
+import { ellipsisNFTName } from '~/util/ui';
+
+const EVENT_TYPE = {
+  COLLECT: 'collect',
+  PUBLISH: 'publish',
+  SEND: 'send',
+  PURCHASE: 'purchase',
+};
+
+const FOLLOW_PROMPT_STATE = {
```
Comment (nwingt) [inline src/components/SocialFeedItem.vue:159]: Should refactor this

File: src/components/SocialFeedItem.vue [hunk 17]
```diff
@@ -0,0 +1,407 @@
...
+      try {
+        this.isFollowPromptUpdating = true;
+        switch (this.followPromptState) {
+          case FOLLOW_PROMPT_STATE.AUTO:
+            this.followPromptState = FOLLOW_PROMPT_STATE.UNFOLLOW;
+            await this.walletUnfollowCreator(followOwnerWallet);
+            logTrackerEvent(
+              this,
+              'NFT',
+              'social_feed_click_unfollow',
```
Comment (nwingt) [inline src/components/SocialFeedItem.vue:374]:
```suggestion
              'FeedUnfollowClick',
```

File: src/components/SocialFeedItem.vue [hunk 18]
```diff
@@ -0,0 +1,407 @@
...
+            );
+            break;
+          case FOLLOW_PROMPT_STATE.UNFOLLOW:
+          default:
+            this.followPromptState = FOLLOW_PROMPT_STATE.AUTO;
+            await this.walletFollowCreator(followOwnerWallet);
+            logTrackerEvent(
+              this,
+              'NFT',
+              'social_feed_click_follow',
```
Comment (nwingt) [inline src/components/SocialFeedItem.vue:386]:
```suggestion
              'FeedFollowClick',
```

**Review by nwingt — COMMENTED** [review]
I think we should rename `social-feed` route to just `feed`

**Pull request merged**